squad/ci/models.py

```diff
@@ -257,6 +257,12 @@ def force_resubmit(self):
     def cancel(self):
         if self.job_id is not None:
             return self.backend.get_implementation().cancel(self)
+        else:
```

Comment:
What do you think about making this block of code available even if job_id exists? This way job_status would turn to "Cancelled" right away.

Reply:
I don't think it's a good idea. You can imagine a situation (assuming LAVA backend) when you add a job with watch_job API and then try to cancel it but you don't have enough privileges. You can view the job, but not alter it. In such case job_status would be incorrect. I think it only applies to jobs that haven't been yet submitted, thus no job_id.